packages/storage/amplify_storage_s3_dart/lib/src/platform_impl/download_file/download_file_html.dart

```diff
@@ -0,0 +1,97 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:amplify_core/amplify_core.dart';
+import 'package:amplify_storage_s3_dart/amplify_storage_s3_dart.dart';
+import 'package:amplify_storage_s3_dart/src/platform_impl/download_file/dom_helper.dart';
+import 'package:amplify_storage_s3_dart/src/storage_s3_service/storage_s3_service.dart';
+
+/// The html implementation of `downloadFile` API.
+S3DownloadFileOperation downloadFile({
+  required StorageDownloadFileRequest request,
+  required S3PluginConfig s3pluginConfig,
+  required StorageS3Service storageS3Service,
+  required AppPathProvider appPathProvider,
+  void Function(S3TransferProgress)? onProgress,
+}) {
+  Future<void> noOp() async {}
+  return S3DownloadFileOperation(
+    request: StorageDownloadFileRequest(
+      key: request.key,
+      localFile: request.localFile,
+      options: request.options as S3DownloadFileOptions?,
+    ),
+    result: _downloadFromUrl(
+      request: request,
+      s3pluginConfig: s3pluginConfig,
+      storageS3Service: storageS3Service,
+    ),
+    // In Web the download process is managed by the browser so the operation
+    // control APIs do nothing here.
+    resume: noOp,
+    pause: noOp,
+    cancel: noOp,
+  );
+}
+
+Future<S3DownloadFileResult> _downloadFromUrl({
+  required StorageDownloadFileRequest request,
+  required S3PluginConfig s3pluginConfig,
+  required StorageS3Service storageS3Service,
+}) async {
+  const defaultExpiresIn = Duration(minutes: 5);
```

Comment:
This should be the default param in `S3GetUrlOptions`

Reply:
Yea that should be better.
Will update the default expires to 15 mins align with amplify-js (while amplify-ios is having default value 5 hrs and and amplify-android sis having default value 7 days for `getUrl`...)

Reply:
Updated the default `expires` in https://github.com/aws-amplify/amplify-flutter/pull/2338/commits/58ca17be9e92b34f3a180b0a2e653470a9c97943